### PR TITLE
feat: add event informer, collect memory-pressure evicted events

### PR DESCRIPTION
**Pull request by UndeadRat22**

### CODE REVIEW

File: internal/services/controller/handlers/filters/onlypodoom/filter.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package onlypodoom
```
Comment (dariuspie) [inline internal/services/controller/handlers/filters/onlypodoom/filter.go:1]: Sounds a bit complex for package name. Maybe 'oomevents' would fit?

File: internal/services/controller/handlers/filters/onlypodoom/filter.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	"castai-agent/internal/services/controller/util"
+)
+
+func Filter(e castai.EventType, obj interface{}) bool {
+	k8sEvent, ok := obj.(*corev1.Event)
+	if !ok {
+		return false
+	}
+
+	return util.IsOOMEvent(k8sEvent)
```
Comment (dariuspie) [inline internal/services/controller/handlers/filters/onlypodoom/filter.go:16]: Here it looks like checking only for OOM event, but inside has more logic. Why is that logic moved to util, if we have a whole package here for that? Would be in single place and clear what it does...

File: internal/services/controller/informers/informer.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	informer cache.SharedInformer,
+	handledType reflect.Type,
+	filters ...filters.Filter,
+) *HandledInformer {
+	log = log.WithField("informer", handledType.String())
+	handler := handlers.NewHandler(log, queue, handledType, filters, defaultTransformers)
+	informer.AddEventHandler(handler)
+
+	return &HandledInformer{
+		SharedInformer: informer,
```
Comment (dariuspie) [inline internal/services/controller/informers/informer.go:39]: nitpick: maybe SharedInformer - simply Informer?

File: internal/services/controller/handlers/filters/onlypodoom/filter.go (hunk 1 above)
Comment (UndeadRat22) [reply internal/services/controller/handlers/filters/onlypodoom/filter.go]: yep, updated

File: internal/services/controller/util/util.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+	ResourceMemory = "memory"
+)
+
+func IsOOMEvent(event *corev1.Event) bool {
+	if event.Reason != ReasonOOMEviction {
+		return false
+	}
+
+	if event.InvolvedObject.Kind != KindPod {
```
Comment (dariuspie) [inline internal/services/controller/util/util.go:23]: Can there be OOM events for non pod kind? Should func name reflect that it checks for more? Unless it is moved to  that filter package.

**Review by dariuspie — APPROVED** [review]

File: internal/services/controller/handlers/filters/onlypodoom/filter.go (hunk 2 above)
Comment (UndeadRat22) [reply internal/services/controller/handlers/filters/onlypodoom/filter.go]: yep, the code was structured differently at the begging of the implementation, kind of a leftover, will remove.

File: internal/services/controller/handlers/filters/onlypodoom/filter.go (hunk 1 above)
Comment (UndeadRat22) [reply internal/services/controller/handlers/filters/onlypodoom/filter.go]: renamed

File: internal/services/controller/informers/informer.go (hunk 3 above)
Comment (UndeadRat22) [reply internal/services/controller/informers/informer.go:39]: The SharedInformer is an embedded struct, renaming this to something else will cause more a.Informer.Method in other locations.

**Review by mindaugasCast — APPROVED** [review]

**Pull request merged**